packages/liferay-js-insights/reporters/json.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+const fs = require('fs');
+const util = require('util');
+
+const writeFile = util.promisify(fs.writeFile);
+
+/**
+ * Filters unwanted dependencies from a module's metadata to simplify the final report.
+ */
+function prepareModule(dependencies) {
+	const simpleModule = {};
+
+	Object.keys(dependencies)
+		.filter(key => dependencies[key] && dependencies[key].length)
+		.forEach(key => {
+			simpleModule[key] = dependencies[key];
+		});
```

Comment:
Pointless micro-optimization; you could do one pass through the keys instead of two, and seeing as you are already using `for`/`of`, we could also just turn off that lint rule in this project and do:

```javascript
for ([key, value] of Object.entries(dependencies)) {
  if (value && value.length) {
    simpleModule[key] = value;
  }
}
```

Don't know if that is more or less readable though. 🤷‍♂

Reply:
Yeah, I've heard of people actually wanting to ban the use of `if`, so I was trying to avoid it :D 

I like how `.filter` reads, and of course, `keys` here is but a minuscule set. On the other hand, we're already mutating something in a `forEach`, so... 🤷‍♂

Reply:
> I've heard of people actually wanting to ban the use of `if`, so I was trying to avoid it 

Radical FP ~enthusiasts~ theoreticians.